pkgs/development/rocm-modules/6/llvm/stage-3/libclc.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenv
+, callPackage
+, rocmUpdateScript
+, llvm
+, clang
+, spirv-llvm-translator
+}:
+
+let
+  spirv = (spirv-llvm-translator.override { inherit llvm; });
+in callPackage ../base.nix rec {
+  inherit stdenv rocmUpdateScript;
+  buildDocs = false; # No documentation to build
+  buildMan = false; # No man pages to build
+  targetName = "libclc";
+  targetDir = targetName;
+  extraBuildInputs = [ spirv ];
+
+  # `spirv-mesa3d` isn't compiling with LLVM 15.0.0, it does with LLVM 14.0.0
+  # Try removing the `spirv-mesa3d` and `clspv` patches next update
+  # `clspv` tests fail, unresolved calls
+  extraPostPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace "find_program( LLVM_CLANG clang PATHS \''${LLVM_BINDIR} NO_DEFAULT_PATH )" \
+        "find_program( LLVM_CLANG clang PATHS \"${clang}/bin\" NO_DEFAULT_PATH )" \
+      --replace "find_program( LLVM_SPIRV llvm-spirv PATHS \''${LLVM_BINDIR} NO_DEFAULT_PATH )" \
+        "find_program( LLVM_SPIRV llvm-spirv PATHS \"${spirv}/bin\" NO_DEFAULT_PATH )" \
+      --replace "  spirv-mesa3d-" "" \
+      --replace "  spirv64-mesa3d-" "" \
+      --replace "NOT \''${t} MATCHES" \
+        "NOT \''${ARCH} STREQUAL \"clspv\" AND NOT \''${ARCH} STREQUAL \"clspv64\" AND NOT \''${t} MATCHES"
+  '';
+
+  checkTargets = [ ];
+  isBroken = true; # ROCm 5.7.0 doesn't have IR/AttributeMask.h yet...?
```

Comment:
Should this still be marked as broken?

Reply:
For all of those broken packages I have at least tried to build them, and if they have their own `src` checked if it is abandoned.

For this, besides the one missing header file, the build yells a lot about `CapabilityLongConstantCompositeINTEL`:

```
SPIRV-LLVM-Translator> /build/source/lib/SPIRV/SPIRVReader.cpp:62:10: fatal error: llvm/IR/AttributeMask.h: No such file or directory
SPIRV-LLVM-Translator>    62 | #include "llvm/IR/AttributeMask.h"
SPIRV-LLVM-Translator>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~
SPIRV-LLVM-Translator> compilation terminated.
SPIRV-LLVM-Translator> make[2]: *** [lib/SPIRV/CMakeFiles/LLVMSPIRVLib.dir/build.make:314: lib/SPIRV/CMakeFiles/LLVMSPIRVLib.dir/SPIRVReader.cpp.o] Error 1
SPIRV-LLVM-Translator> make[2]: *** Waiting for unfinished jobs....
SPIRV-LLVM-Translator> /build/source/lib/SPIRV/SPIRVWriter.cpp:5531:5: warning: multi-line comment [-Wcomment]
SPIRV-LLVM-Translator>  5531 |     // <id>ResTy Res<id> In<id> \
SPIRV-LLVM-Translator>       |     ^
SPIRV-LLVM-Translator> In file included from /build/source/lib/SPIRV/SPIRVInternal.h:45,
SPIRV-LLVM-Translator>                  from /build/source/lib/SPIRV/OCLUtil.h:42,
SPIRV-LLVM-Translator>                  from /build/source/lib/SPIRV/SPIRVToOCL20.cpp:40:
SPIRV-LLVM-Translator> /build/source/lib/SPIRV/libSPIRV/SPIRVNameMapEnum.h: In member function 'void SPIRV::SPIRVMap<Ty1, Ty2, Identifier>::init() [with Ty1 = spv::Capability; Ty2 = std::__cxx11::basic_string<char>; Identifier = void]':
SPIRV-LLVM-Translator>   613 |   add(CapabilityLongConstantCompositeINTEL, "LongConstantCompositeINTEL");
SPIRV-LLVM-Translator>       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
SPIRV-LLVM-Translator> In file included from /build/source/lib/SPIRV/libSPIRV/SPIRVType.h:47,
SPIRV-LLVM-Translator>                  from /build/source/lib/SPIRV/SPIRVInternal.h:46:
SPIRV-LLVM-Translator> /build/source/lib/SPIRV/libSPIRV/SPIRVEntry.h: In member function 'SPIRV::SPIRVCapVec SPIRV::SPIRVContinuedInstINTELBase<OC>::getRequiredCapability() const':
SPIRV-LLVM-Translator> /build/source/lib/SPIRV/libSPIRV/SPIRVEntry.h:911:19: error: 'CapabilityLongConstantCompositeINTEL' was not declared in this scope
SPIRV-LLVM-Translator>   911 |     return getVec(CapabilityLongConstantCompositeINTEL);
```

maybe that's related to https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/2261